src/Symfony/Component/Process/Tests/Messenger/RunProcessMessageHandlerTest.php

```diff
@@ -44,4 +44,31 @@ public function testRunFailedProcess()
 
         $this->fail('Exception not thrown');
     }
+
+    public function testRunSuccessfulProcessFromShellCommandline()
+    {
+        $context = (new RunProcessMessageHandler())(RunProcessMessage::fromShellCommandline('ls | grep Test', cwd: __DIR__));
+
+        $this->assertSame('ls | grep Test', $context->message->commandLine);
+        $this->assertSame(0, $context->exitCode);
+        $this->assertStringContainsString(basename(__FILE__), $context->output);
+    }
+
+    public function testRunFailedProcessFromShellCommandline()
+    {
+        try {
+            (new RunProcessMessageHandler())(RunProcessMessage::fromShellCommandline('invalid'));
+        } catch (RunProcessFailedException $e) {
+            $this->assertSame('invalid', $e->context->message->commandLine);
+            $this->assertContains(
+                $e->context->exitCode,
+                [null, '\\' === \DIRECTORY_SEPARATOR ? 1 : 127],
+                'Exit code should be 1 on Windows, 127 on other systems, or null',
+            );
+
+            return;
+        }
+
+        $this->fail('Exception not thrown');
```

Comment:
the code is strange already but let's make the new one cleaner :)

```suggestion
            $this->fail('Exception not thrown');
        } catch (RunProcessFailedException $e) {
            $this->assertSame('invalid', $e->context->message->commandLine);
            $this->assertContains(
                $e->context->exitCode,
                [null, '\\' === \DIRECTORY_SEPARATOR ? 1 : 127],
                'Exit code should be 1 on Windows, 127 on other systems, or null',
            );
        }
```

Reply:
Done :+1: